ISO 7379 screws

slowinternets opened the pull request:
I needed these screws for a project, so I added them to the workbench.

shaise commented:
Hi!
Thanks for your contribution.
Is it possible for you to fix the following issues:
1. There is an error when i switch 'thread' property to 'true'. It is OK not to support true thread for the moment, but you need to prevent it from generating an error.
2. Add the hex socket. (the socket looks like a cone):
![newbolt](https://user-images.githubusercontent.com/5813568/82136783-c41c1300-9819-11ea-8b5a-6a3d7ca4b518.png)
Thanks!
shai

slowinternets commented:
Well, that was interesting. 
I originally tested my changes on a V0.19 Linux install of FreeCAD, where everything worked as intended:
![thread_true_V019_linux](https://user-images.githubusercontent.com/37948669/82156216-32230300-9837-11ea-9a4c-1dadfdb862bb.png)
I was only able to reproduce the errors you showed after testing on a V0.18 install (on windows, though I don't think the OS is relevant.)

I fixed some faulty calls to the Part module. I also changed the hex-socket generating step to not rely on FreeCAD to enumerate shape  faces in a certain order. 

Here is how it looks on my V0.18 install now:
![V018_changes_made](https://user-images.githubusercontent.com/37948669/82156544-d2c5f280-9838-11ea-874b-7e7161873761.png)
Thanks,
Alex.

shaise commented:
Thanks a lot! its working perfect now.
I know most of us use 0.19 now, but 0.18 is still considered the stable version, so it is important to support it as well.
